ignition/locations/kubernetes.py

```diff
@@ -0,0 +1,168 @@
+            should_remove = temp_only == False or config_file['is_temp']
+            if should_remove and os.path.exists(config_file['path']):
+                os.remove(config_file['path'])
+            else:
+                remaining_config_files.append(config_file)
+        self.config_files_created = remaining_config_files
+        
+    def to_dict(self):
+        """
+        Produces a dictionary copy of the deployment location
+
+        Returns: 
+            the deployment location configuration as a dictionary. For example:
+            
+            {
+                'name': 'Test',
+                'properties': {
+                    'client_config': { ... },
+                    'default_object_namespace': 'default'
+                }
+            }
+        """
+        return {
+            KubernetesDeploymentLocation.NAME: self.name,
+            KubernetesDeploymentLocation.PROPERTIES: {
+                KubernetesDeploymentLocation.CONFIG_PROP: self.client_config,
+                KubernetesDeploymentLocation.DEFAULT_OBJECT_NAMESPACE_PROP: self.default_object_namespace
+            }
+        }
+
+class KubernetesConfigValidationError(Exception):
+    pass
+
+class KubernetesSingleConfigValidator:
+    """
+    Validates Kubernetes configuration is for a single cluster
+    """
+
+    @staticmethod
+    def validate(config):
+        KubernetesSingleConfigValidator(config).run_validation()
+
+    def __init__(self, config):
+        self.config = config
+
+    def run_validation(self):
+        cluster = self.__validate_single_named_item('clusters')
+        user = self.__validate_single_named_item('users')
+        context = self.__validate_single_named_item('contexts')
+        self.__validate_context_is_for_cluster(context, cluster)
+        self.__validate_context_is_for_user(context, user)
+        self.__validate_context_is_current(context)
+
+    def __validate_single_named_item(self, in_key):
+        if in_key not in self.config:
+            raise KubernetesConfigValidationError(f'Config missing \'{in_key}\'')
+        values = self.config.get(in_key)
+        if not isinstance(values, list):
+            raise KubernetesConfigValidationError(f'Config item \'{in_key}\' expected to be a list but was \'{type(values).__name__}\'')
+        num_of_values = len(values)
+        if num_of_values != 1:
+            raise KubernetesConfigValidationError(f'Config item \'{in_key}\' expected to be a list of one but there are {num_of_values} elements')
+        if 'name' not in values[0]:
+            raise KubernetesConfigValidationError(f'Config item \'{in_key}[0]\' missing \'name\'')
+        return values[0]
+
+    def __validate_context_is_for_cluster(self, context, cluster):
+        context_name = context.get('name')
+        context_details = context.get('context')
+        context_cluster = context_details.get('cluster', None)
+        cluster_name = cluster.get('name')
```

Comment:
default is None, so is this needed - same for below too

Reply:
sorry I don't understand what you mean? The default is None because we haven't checked that key is there before.

Reply:
Sorry, probably didn't explain very well. I'm just saying that the default value for the "default" parameter for dict.get is None isn't it, so is the explicit "None" required?

Reply:
ah ok I'm with you now